### PR TITLE
various vulnerability fixes

### PR DESCRIPTION
**Pull request by vilay-nference**

### CODE REVIEW

File: Dockerfile.nfer [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Use the specified base image
+FROM hz-registry.nferx.com/ubuntu as builder
```
Comment (jonn-smith) [inline Dockerfile.nfer:2]: @vilay-nference Where is the dockerfile for this image?  It is not immediately accessible, so we cannot verify its contents.  

In general, we use base images provided directly from first-party sources (e.g. `https://hub.docker.com/_/ubuntu`).  

Please change this to use a standard base image and then we can re-evaluate.

File: build.gradle [hunk 2]
```diff
@@ -232,6 +239,12 @@ configurations {
 }
 
 dependencies {
+    implementation('net.minidev:json-smart:2.4.9') {
```
Comment (lbergelson) [inline build.gradle:242]: Could you explain what these new dependencies are?

File: Dockerfile.nfer [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Use the specified base image
```
Comment (droazen) [inline Dockerfile.nfer:1]: Remove this third-party Dockerfile

File: build.gradle [hunk 4]
```diff
@@ -232,6 +239,12 @@ configurations {
...
+        exclude group: 'net.minidev', module: 'json-smart'
+    }
+    // Example dependencies
+    implementation 'biz.aQute.bnd:biz.aQute.bndlib:5.1.2'
```
Comment (droazen) [inline build.gradle:246]: Don't add new GATK dependencies here unless absolutely necessary

File: Dockerfile.nfer (hunk 1 above)
Comment (vilay-nference) [reply Dockerfile.nfer:2]: This is used for internal purpose. I'll delete this file .. not required for gatk

File: build.gradle (hunk 2 above)
Comment (vilay-nference) [reply build.gradle:243]: This is used as transitive dependencies in one of the package. Overriding the newer version for the compatibles.

File: Dockerfile.nfer (hunk 3 above)
Comment (vilay-nference) [reply Dockerfile.nfer:1]: Yes. will remove this and update you

File: build.gradle (hunk 4 above)
Comment (vilay-nference) [reply build.gradle:250]: Again dnsjava is used from hadoop client.

File: Dockerfile.nfer (hunk 3 above)
Comment (vilay-nference) [reply Dockerfile.nfer]: This is deleted

File: build.gradle (hunk 4 above)
Comment (vilay-nference) [reply build.gradle:250]: @droazen . Let us know if any more modifications required

**Comment by lbergelson** [comment]
@vilay-nference Thank you for doing this work.  It's nitpicky annoying stuff to figure out.
 
I have one additional request.   Instead of addding additional direct implementation dependencies, could we specify the transtive version requirements in a [gradle constraints block](https://docs.gradle.org/current/userguide/dependency_constraints.html)? 

That will: 
 1. make it clear that we don't rely on these directly
 2.  prevent us from keeping them around if we do something like remove hadoop in the future
 3.  lets us rewrite those force blocks to instead define minimum versions so if the libraries move forward in the future we're not accidentally holding on a to an old version

**Comment by lbergelson** [comment]
@vilay-nference Were you able to get this configuration to pass tests on your end?  I've attempted to incorporate your changes into https://github.com/broadinstitute/gatk/pull/8998, but I'm running into issues with hadoop and protobuf incompatibilities.  I see the same problem with your branch when I try to run tests on it.  (I also can't run tests without disabling -Werror on your branch since there are still some unresolved deprecation and other minor issues).

Errors look like this:

```
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.security.proto.SecurityProtos [in thread "IPC Server handler 1 on default port 64812"]
	at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos.<clinit>(ClientNamenodeProtocolProtos.java)
```

and you can easily trigger one by running `ParallelCopyGCSDirectoryIntoHDFSSparkIntegrationTest`

**Comment by lbergelson** [comment]
I've incorporated the the changes from here into #9006 which is now merged.  Thank you @vilay-nference

**Pull request closed without merging**